Tribler/Test/test_tribler_profile.py

```diff
@@ -0,0 +1,23 @@
+import os
+import tempfile
+
+from Tribler.Main.tribler_profiler import run_tribler_with_yappi
+from Tribler.Test.test_as_server import AbstractServer
+
+
+class TestTriblerProfile(AbstractServer):
+    def test_profile_yappi_environment_set(self):
+        temp_dir = tempfile.mkdtemp()
+        os.environ['YAPPI_OUTPUT_DIR'] = temp_dir
+
+        def a_simple_loop():
+            for i in range(0, 10):
+                pass
+
+        import Tribler.Main.tribler_profiler as tribler_file
+        old_func_pointer = tribler_file.run_tribler
+        tribler_file.run_tribler = a_simple_loop
+        run_tribler_with_yappi()
+
+        self.assertTrue(os.path.exists(os.path.join(temp_dir, 'yappi.callgrind')))
+        tribler_file.run_tribler = old_func_pointer
```

Comment:
Bad idea, if the test crashes, you never restore the hijacked function.

This kind of things should be avoided or moved to setup/teardown.

In this case, just make an optional argument so you can pass which function to profile to `run_tribler_with_yappi()`.